lib/parsers/aiff_parser.rb

```diff
@@ -18,8 +18,20 @@ class FormatParser::AIFFParser
     'ANNO',
   ]
 
-  def information_from_io(io)
+  NATURES = [:audio].freeze
+  FORMATS = [:aiff].freeze
+
+  def self.natures
+    NATURES
+  end
+
+  def self.formats
```

Comment:
See above. Why don't we let the parser objects tell us which formats they are willing to accept after they are instantiated? Also, who not just return a symbol - i.e. why do we need constants - that we also have to get right in every parser? Moreover - what if we let every parser answer a question as opposed to returning a hardcoded list:

```
def will_emit_format?(some_format)
  ...
end
```

Then you can keep any allocations you might have to do for this within the method. It will also simplify your `select()` in the main `parse()` method  (`accepted_natures.any?{|nature| parser.will_emit?(nature) }`).

Computationally it will be slightly more overhead but this is peanuts compared to the actual grunt work parsers have to do.

Reply:
Is there any reason why you want this at the instance level and not in the class level? However, I like the idea of a question method. That can make the code more readable even though it adds some degree of code duplication, but nothing crazy that can't be assumed.

Reply:
The reason is that it's easier to write a parser implementation then - the mental model of a parser is going to be simpler, you won't have to remember where you need class methods and where you need instance methods.

Reply:
Done.